File: c++/2551.cpp
#include<iostream>
using namespace std;

int main () {
    float T, D = 0;
    int x, treinos;
    float record;
    record = 0;
    while (cin>>treinos)
        for (int dias = 1; dias < treinos+1; dias++)
        {
            cin >> T >> D;
            float recordAtual = D/T;
            if (recordAtual > record)
            {
                cout << dias << endl;
                record = recordAtual;
            }
        }
    return 0;
}

File: python/2551.py
while 1:
    try:
        treinos = int(input())
        recorde = 0
        for dias in range(treinos):
            T, D = map(int, input().split())
            recordeAtual = D/T
            if recordeAtual > recorde:
                recorde = recordeAtual
                print(dias+1)
    except EOFError:
        break